File: 每日一题/94. 二叉树的中序遍历.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        stk = []
        ret = []
        while root or len(stk) > 0:
            while root:
                stk.append(root)
                root = root.left
            root = stk.pop()
            ret.append(root.val)
            root = root.right

        return ret

File: 每日一题/94. 二叉树的中序遍历.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution
{
public:
    vector<int> ret;
    void inOrder(TreeNode *root)
    {
        if (root == NULL)
        {
            return;
        }

        inOrder(root->left);
        ret.push_back(root->val);
        inOrder(root->right);
    }
    vector<int> inorderTraversal(TreeNode *root)
    {
        inOrder(root);
        return ret;
    }
};